### PR TITLE
Cleanup `Theme`-related Code

### PR DESCRIPTION
**Pull request by Nerixyz**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

I mentioned this in #4447 at the beginning a bit: The theme code is hard to read and hard to maintain. This PR attempts to clean this code up (a bit at least). If you see more clean-up possibilities, let me know.

The commits should be reviewable on their own.

### CODE REVIEW

File: src/singletons/Theme.cpp [hunk 1]
```diff
@@ -66,216 +48,152 @@ Theme::Theme()
...
-        this->splits.dropTargetRect = QColor(0, 148, 255, 0x00);
-        this->splits.dropTargetRectBorder = QColor(0, 148, 255, 0x00);
-
-        this->splits.resizeHandle = QColor(0, 148, 255, 0x70);
-        this->splits.resizeHandleBackground = QColor(0, 148, 255, 0x20);
-    }
-
-    this->splits.header.background = getColor(0, sat, flat ? 1 : 0.9);
-    this->splits.header.border = getColor(0, sat, flat ? 1 : 0.85);
+    this->messages.selection = QColor::fromHsl(0, 0, isLight ? 0 : 255, 54);
```
Comment (pajlada) [inline src/singletons/Theme.cpp:163]: Undo the change to `this->message.selection`

File: src/singletons/Theme.cpp [hunk 2]
```diff
@@ -66,216 +48,152 @@ Theme::Theme()
...
+
+    // Splits
+    this->splits.dropTargetRect = QColor(0, 148, 255, 0);
+    this->splits.dropTargetRectBorder = QColor(0, 148, 255, 0);
```
Comment (pajlada) [inline src/singletons/Theme.cpp:167]: Let's keep the light/dark theme colors of `dropTargetRect` even if their alpha is 0

File: src/singletons/Theme.cpp [hunk 3]
```diff
@@ -66,216 +48,152 @@ Theme::Theme()
...
-
-    this->splits.input.focusedLine = this->tabs.highlighted.line.regular;
+        (isLight ? "#68B1FF" : this->tabs.selected.backgrounds.regular.name());
 
     this->splits.messageSeperator =
-        isLight ? QColor(127, 127, 127) : QColor(60, 60, 60);
-    this->splits.background = getColor(0, sat, 1);
-    this->splits.dropPreview = QColor(0, 148, 255, 0x30);
-    this->splits.dropPreviewBorder = QColor(0, 148, 255, 0xff);
+        QColor::fromHslF(0, 0, isLight ? 0.5 : 0.24);
```
Comment (pajlada) [inline src/singletons/Theme.cpp:192]: Undo change to `messageSeperator`

**Review by pajlada — CHANGES_REQUESTED** [review]
Small nitpick, overall looks good

**Review by pajlada — APPROVED** [review]

**Review by Mm2PL — APPROVED** [review]
LGTM

**Pull request merged**